### DISCUSSION TITLE
AEADChaCha20Poly1305 encrypt with associated_data

### DISCUSSION THREAD

**Discussion by GreenSnakeLinux** [krzyzanowskim/CryptoSwift#952] [General]
Hello,
In Python there is the following class in cryptography\hazmat\primitives\ciphers\aead.py

class ChaCha20Poly1305:
def encrypt(
	self,
	nonce: bytes,
	data: bytes,
	associated_data: typing.Optional[bytes],
) -> bytes:

It takes an additional parameter: associated_data that is not present in your swift implementation:
let encrypt = try AEADChaCha20Poly1305.encrypt(plaintext, key: key, iv: nonce, authenticationHeader: header)
(It's not authenticationHeader, my test indicate that only the last 16 bytes are impacted)

Is it a miss in the implementation or there is a way to do the same as Python ?

thanks

**Comment by krzyzanowskim** [comment]
It's missing API to do so. The schema is AEAD however we don't expose any API to attach data to it.
